tensorflow/core/graph/mkl_optimizer_merge.h

```diff
@@ -30,188 +30,13 @@ limitations under the License.
-                         // from this node. If mfwd_ is farther than mmaxhop_
-                         // then we do not rewrite the node.
-    } ContextInfo;
-    ContextInfo cinfo;  // Context for rewrite
-  } RewriteInfo;
-
-  /// Structure to store all constant strings
-  typedef struct {
-    string conv2d;
-    string conv2dwithbias;
-    string conv2dwithbiasbackpropbias;
-    string biasadd;
-
-    string matmul;
-    string matmulmkl;
-    string add;
-
-    string biasaddgrad;
-  } ConstStringInfo;
-
-  ConstStringInfo csinfo_;
-  std::vector<MergeInfo> minfo_;
-  std::vector<RewriteInfo> rinfo_;
-
- private:
-  /*
-   *  Return a node that can be merged with input node
-   *
-   *  @return pointer to the node if we can find such a
-   *  node. Otherwise, it returns NULL.
-   */
-  Node* FindNodeForMerge(const Node* a) const;
-
-  /*
-   *  Merge predecessor node with its successor.
-   *  Currently, we support merging Conv2D with AddBias, and
-   *  MatMul with Add node.
-   *
-   *  Input nodes succ and pred may be deleted if the call to
-   *  this function is successful. Attempt to use the pointers
-   *  after the call to function may result is undefined behaviors.
-   *
-   *  @input g - input graph, succ - successor node, pred - predecessor node
-   *  @return true, if merging is successful and supported.
-   *           Returns false otherwise.
-   */
-  bool MergeNode(std::unique_ptr<Graph>* g, Node *succ, Node *pred);
-
-  /*
-   *  Can the input node (n) be rewritten via rewrite node method?
-   *
-   *  @return true, if it can be rewritten; false, otherwise.
-   *          In case of true, returns context ID of the matching context.
-   */
-  bool IsApplicableRewriteNode(const Node *n) const;
-
-  /*
-   * Rewrite input node to its corresponding node specified in rewrite info.
-   *
-   * Input node may be deleted in case of rewrite. Attempt to use the node
-   * after the call can result in undefined behaviors.
-   *
-   * @input  g - input graph, n - Node to be rewritten
-   * @return true, if the input node can be rewritten; false, otherwise.
-   *         Graph is updated in case the input node can be rewritten.
-   *         Otherwise, it is not updated.
-   */
-  bool RewriteNode(std::unique_ptr<Graph>* g, Node *n);
-
-  /*
-   * Helper function that searches the matching rewriteinfo for the node.
-   * Implements depth-first search in the data dependence graph for the
-   * gradient op in backward direction.
-   *
-   * @input n - Node (gradient op) whose rewriteinfo is to be searched,
-   *        fwdn - pointer to node from the forward pass that this node
-   *        belongs to
-   * @return Matching rewriteinfo in case a match is found; null otherwise.
-   */
-  const RewriteInfo* FindMatchingRewriteInfo(const Node *n,
-                                             const Node **fwdn) const;
-};
-
-/*
- * Interface to invoke the pass for unit test
- */
+// Returns true if and only if 'g' is mutated.
```

Comment:
You could put a little wrapper class in the header and still leave most of the implementation in the .cc file. If you choose to keep things as they are, then please prefix the method below with Test_ if it is to be used only by unit tests.